frame/bounties/src/lib.rs

```diff
@@ -653,12 +753,628 @@ decl_module! {
+
+						T::Currency::ensure_can_withdraw(
+							&bounty_account,
+							expect_balance,
+							WithdrawReasons::TRANSFER,
+							expect_free_balance,
+						).map_err(|_| Error::<T>::InsufficientBountyBalance)?;
+
+						// Use bounty counter to generate
+						// subbounty id
+						let subbounty_id = Self::bounty_count();
+
+						// Increment the active subbounty count.
+						bounty.active_subbounty_count += 1;
+
+						BountyCount::put(subbounty_id + 1);
+
+						// Transfer fund from parent bounty to subbounty.
+						let subbounty_account = Self::bounty_account_id(subbounty_id);
+						let _ = T::Currency::transfer(
+							&bounty_account,
+							&subbounty_account,
+							value,
+							AllowDeath
+						); // should not fail
+
+						// Create subbounty instance
+						Self::create_subbounty(
+							bounty_id,
+							subbounty_id,
+							description,
+							value
+						);
+						Ok(())
+					} else {
+						Err(Error::<T>::UnexpectedStatus.into())
+					}
+				}
+			)?;
+		}
+
+		/// Propose subcurator for funded subbounty.
+		///
+		/// The dispatch origin for this call must be master curator.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Proposed subcurator may be "master curator",
+		/// and subcurator fee & curator deposit
+		/// may be Zero in this case.
+		///
+		/// Subbounty must be in "Funded" state, for
+		/// processing the call. and state of subbounty is
+		/// moved to CuratorProposed on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		/// - `subcurator`: Address of subcurator.
+		/// - `fee`: payment fee to subcurator for execution.
+		#[weight = 10_000]
+		fn propose_subcurator(origin, #[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+			subcurator: <T::Lookup as StaticLookup>::Source,
+			#[compact] fee: BalanceOf<T>,
+		) {
+			let signer = ensure_signed(origin)?;
+			let subcurator = T::Lookup::lookup(subcurator)?;
+
+			// Ensure parent bounty is Active & get status of curator
+			let (master_curator, _) = Self::ensure_bounty_active(bounty_id)?;
+
+			// Mutate the Subbounty instance
+			SubBounties::<T>::try_mutate_exists(
+				bounty_id,
+				subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+
+					let mut subbounty = maybe_subbounty
+						.as_mut()
+						.ok_or(Error::<T>::InvalidIndex)?;
+
+					// Ensure sure caller is curator
+					ensure!(signer == master_curator, Error::<T>::RequireCurator);
+
+					// Ensure subbounty is in expected state
+					ensure!(
+						subbounty.status == SubBountyStatus::Added,
+						Error::<T>::UnexpectedStatus
+					);
+
+					// Ensure subcurator fee is less than subbounty value.
+					ensure!(fee < subbounty.value, Error::<T>::InvalidFee);
+
+					// Update the master curator fee balance.
+					Bounties::<T>::mutate_exists(
```

Comment:
May be a question, for me it occurs safe operation because
existence of bounty object is ensured with help of below expression.

```
let (master_curator, _) = Self::ensure_bounty_active(bounty_id)?;
```

So mutation comes after this expression, is there sill possibility for error ?

Reply:
You have `ok_or` in the closure. Should change back to `saturating_sub` it you want to avoid error after mutation. 
And should check if the parent bounty fee > sub bounty fee upfront

Reply:
Hello Bryan,

Refactored as below, Plz correct me if I am wrong with my understanding.

```rust
Bounties::<T>::try_mutate_exists(
	bounty_id,
	|maybe_bounty| -> DispatchResult {
		if let Some(bounty) = maybe_bounty.as_mut() {
			// Ensure subcurator fee is less than
			// master curator fee &
			// reduce the master curator fee
			// by subcurator fee.
			ensure!(fee < bounty.fee, Error::<T>::InvalidFee);
			bounty.fee = bounty
				.fee
				.saturating_sub(fee);
		}
		Ok(())
	}
)?;
```